app/ports/views.py

```diff
@@ -148,27 +158,47 @@ def portdetail_build_information(request):
 
 
 def portdetail_stats(request):
+    days = request.GET.get('days', 30)
+    days_ago = request.GET.get('days_ago', 0)
+    try:
+        days = int(days)
+        days_ago = int(days_ago)
+    except ValueError:
+        return HttpResponse("'days' and 'days-ago' must be integer.")
+    allowed_days = [0, 7, 30, 90, 180, 365]
+    if days not in allowed_days or days_ago not in allowed_days:
+        return HttpResponse("'days' and 'days_ago' can accept only these values: {}".format(allowed_days))
     port_name = request.GET.get('port_name')
     port = Port.objects.get(name__iexact=port_name)
 
-    # Section for calculation of current stats
-    submissions_last_30_days = Submission.objects.filter(timestamp__gte=datetime.datetime.now(tz=datetime.timezone.utc) - datetime.timedelta(days=30)).order_by('user', '-timestamp').distinct('user')
-    installations_last_30_days = PortInstallation.objects.filter(submission_id__in=Subquery(submissions_last_30_days.values('id')), port__iexact=port_name)
-    requested_count = installations_last_30_days.filter(requested=True).values('port').aggregate(Count('port'))
-    total_count = installations_last_30_days.values('port').aggregate(Count('port'))
-    version_distribution = installations_last_30_days.values('version').annotate(num=Count('version')).order_by('-num')
-    os_distribution = installations_last_30_days.values('submission__os_version').annotate(num=Count('submission__os_version')).order_by('-num')
+    end_date = datetime.datetime.now(tz=datetime.timezone.utc) - datetime.timedelta(days_ago)
```

Comment:
What is `datetime.timedelta(days_ago)`? What time unit?

Reply:
I believe the proper syntax would be:
```python
datetime.timedelta(days=days_ago)
```